proxy.go

```diff
@@ -37,8 +37,15 @@ type ProxyHandler struct {
 
 type proxyFunc func(*http.Request) (*url.URL, error)
 
-func NewProxyHandler(proxy proxyFunc, auth *authenticator, block func(string)) ProxyHandler {
-	return ProxyHandler{&http.Transport{Proxy: proxy}, auth, block}
+func NewProxyHandler(auth *authenticator, block func(string)) ProxyHandler {
+	getProxyFromContext := func(req *http.Request) (*url.URL, error) {
```

Comment:
Wondering why this is a named closure instead of just a plain function, as it does not close over any variables. You could put the stand-alone function in `proxyfinder.go` alongside the code that adds the value to the context.

Reply:
It's been a while since I wrote this (it's been sitting on a local branch since last year). I *think* the reason was that it never gets called from anywhere else; it only ever gets called when someone calls the `Transport`'s `Proxy` function. So basically an attempt to minimise the scope of this symbol.

That said, I don't really like how the proxy is added and taken from the context in different middleware layers, so I've gone with your suggestion of moving it into `proxyfinder.go`.